src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java

```diff
@@ -0,0 +1,50 @@
+package com.exadel.discountwebapp.vendor.mapper;
+
+import com.exadel.discountwebapp.location.entity.Location;
+import com.exadel.discountwebapp.location.service.LocationService;
+import com.exadel.discountwebapp.vendor.entity.Vendor;
+import com.exadel.discountwebapp.vendor.vo.VendorRequestVO;
+import com.exadel.discountwebapp.vendor.vo.VendorResponseVO;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+public class VendorMapper {
+    private LocationService locationService;
+
+    public Vendor toEntity(VendorRequestVO request) {
+
+        Location location = locationService.findEntityById(request.getLocationId());
+
+        return Vendor.builder()
+                .title(request.getTitle())
+                .description(request.getDescription())
+                .imageUrl(request.getImageUrl())
+                .email(request.getEmail())
+                .location(location)
+                .build();
+    }
+
+    public VendorResponseVO toResponseVO(Vendor vendor) {
+        return VendorResponseVO.builder()
+                .id(vendor.getId())
+                .title(vendor.getTitle())
+                .description(vendor.getDescription())
+                .imageUrl(vendor.getImageUrl())
+                .email(vendor.getEmail())
+                .locationId(vendor.getLocation().getId())
+                .build();
+    }
+
+    public Vendor updateVO(Vendor vendor, VendorRequestVO request) {
+        Location location = locationService.findEntityById(request.getLocationId());
+
+        vendor.setTitle((request.getTitle()));
```

Comment:
redundant brackets

Reply:
you notice everything)

Reply:
if we have @transactional, there by default propagation= Propagation. REQUIRED. 
I have changed for similar come everywhere.
I have read about this propagation required it apply transaction when physical one happens.
But they also mention about logical one, can't get what is it, logical transaction. 
Could you help me to clear this question

Reply:
got it - logical transaction its "chainy" code calling without real data tranansfer